app/models/work_activity.rb

```diff
@@ -39,34 +45,82 @@ def users_referenced
   end
 
   def created_by_user
-    return nil if created_by_user_id.nil?
+    return nil unless created_by_user_id
+
     User.find(created_by_user_id)
   end
 
-  def message_html
-    if activity_type == "CHANGES"
+  def self.unknown_user
+    "Unknown user outside the system"
+  end
+
+  def comment_event_type?
+    activity_type == COMMENT
+  end
+
+  def changes_event_type?
+    activity_type == CHANGES
+  end
+
+  def file_changes_event_type?
+    activity_type == FILE_CHANGES
+  end
+
+  def system_event_type?
+    activity_type == SYSTEM
+  end
+
+  def log_event_type?
+    system_event_type? || file_changes_event_type? || changes_event_type?
+  end
+
+  def event_type
+    if comment_event_type?
+      "comment"
+    else
+      "log"
+    end
+  end
+
+  def to_html
+    if changes_event_type?
       metadata_changes_html
-    elsif activity_type == "FILE-CHANGES"
+    elsif file_changes_event_type?
       file_changes_html
     else
       comment_html
     end
   end
+  alias message_html to_html
 
   private
 
-    def comment_html
-      # convert user references to user links
-      text = message.gsub(USER_REFERENCE) do |at_uid|
-        uid = at_uid[1..-1]
-        user = User.where(uid: uid).first
-        user_info = user&.display_name_safe || uid
-        "<a class='comment-user-link' title='#{user_info}' href='{USER-PATH-PLACEHOLDER}/#{uid}'>#{at_uid}</a>"
-      end
-      # allow ``` for code blocks (Kramdown only supports ~~~)
-      text = text.gsub("```", "~~~")
-      parsed_document = Kramdown::Document.new(text)
-      parsed_document.to_html
+    def created_at_time
+      created_at.time
+    end
+
+    def event_timestamp
+      created_at_time.strftime("%B %d, %Y %H:%M")
+    end
+
+    def event_timestamp_html
+      "#{event_timestamp} by " if system_event_type? || log_event_type?
+    end
+
+    # This is working
+    def comment_timestamp_html
+      "at #{event_timestamp}" if comment_event_type?
```

Comment:
I'm not sure what `This is working` means. Should there also be a space at the start of the string? If there are going to be paired methods like this, would it be safer to make one `if` and the other `unless`, so we can be confident that precisely one is engaged?